synapse/cortex.py

```diff
@@ -5301,21 +5317,104 @@ async def exportStorm(self, text, opts=None):
                     info.update({'mode': opts.get('mode', 'storm'), 'view': self.iden})
                     self._logStormQuery(text, user, info=info)
 
+                    forms = collections.Counter()
+                    nodec = 0
+                    used_forms = set()
+                    used_types = set()
+                    used_props = set()
+                    used_tagprops = set()
+
                     async for node, path in runt.execute():
+
+                        # Forms
+                        if node.form.isext:
+                            used_forms.add(node.form.name)
+
+                        # Types
+                        form = self.model.forms[node.form.name]
+                        for base in form.type.info.get('bases', ()):
+                            if base.startswith('_'):
+                                used_types.add(base)
+
+                        # Props
+                        for prop in node.form.props.values():
+                            if prop.type.name.startswith('_'):
+                                used_types.add(prop.type.name)
+                            if (node.form.isext or prop.isext) and not prop.name.startswith('.'):
+                                if node.get(prop.name) is not None:
+                                    used_props.add((node.form.name, prop.name))
+
+                        # Tagprops
+                        all_tagprops = node._getTagPropsDict()
+                        for tagname, tagpropdict in all_tagprops.items():
+                            for tagprop, valu in tagpropdict.items():
+                                if valu is not None:
+                                    used_tagprops.add(tagprop)
+
                         await spooldict.set(node.nid, (node.lastlayr(), node.pack()))
+                        forms[node.form.name] += 1
+                        nodec += 1
+
+                    # Edges
+                    ext_edges = set()
+                    for (src_form, verb, tgt_form), edgeinfo in self.model.edges.items():
+                        if verb.startswith('_'):
+                            ext_edges.add((src_form, verb, tgt_form))
```

Comment:
Probably doesn't make too much of a difference but `getExtModel` does return the ext model edges so you could pull them from there rather than filtering the full list.